src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/rest/AuthorizeServlet.java

```diff
@@ -66,7 +66,7 @@ public void authorize(HttpServletRequest request,
         if (ANONYMOUS.equals(userPrincipal)) {
             response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
         } else {
-            ServiceProviderToken newToken = token.authorize(userPrincipal, verifier);
+            ServiceProviderToken newToken = token.authorize(userPrincipal.getName(), verifier);
```

Comment:
I have the same change in my PR. you can target my PR so that we just see your changes

Reply:
done!